src/test/crimson/test_messenger.cc

```diff
@@ -271,10 +271,18 @@ static seastar::future<> test_echo(unsigned rounds,
           return seastar::when_all_succeed(
               // test connecting in parallel, accepting in parallel,
               // and operating the connection reference from a foreign/local core
-              client1->dispatch_pingpong(server1->msgr->get_myaddr(), true),
```

Comment:
@cyx1231st shall we update the comment accordingly? namely, s/foreign//.

and could you update `Client::dispatch_pingpong()` to 

1. remove the default parameter of `foreign_dispatch=true`.
1. move the following comment to there
1. add a `ceph_assert(!foreign_dispatch)` out there.

probably we can have a macro for guarding this unsupported feature, with which we can guard the commented out tests and the proposed assert like:

```c++
#ifndef CRIMSON_MSGR_FOREIGN_DISPATCH
ceph_assert(!foreign_dispatch);
#endif 

// ...
#ifdef CRIMSON_MSGR_CROSS_CORE_DISPATCH
              client1->dispatch_pingpong(server1->msgr->get_myaddr(), true),
              client2->dispatch_pingpong(server2->msgr->get_myaddr(), true),
#endif
```

Reply:
@cyx1231st could you re-consider this comment?

Reply:
Done.